AzureFunctions/EventHubsToElastic/run.csx

```diff
@@ -0,0 +1,151 @@
+    }
+
+    log.Verbose($"Sending Event Hub data to ElasticSearch at {esUri} ...");
+
+    ConnectionSettings connectionSettings = new ConnectionSettings(new Uri(esUri)).BasicAuthentication(username, password);
+    ElasticClient client = new ElasticClient(connectionSettings);
+
+    string currentIndexName = GetIndexName(indexNamePrefix);
+
+    if (!string.Equals(LastIndexName, currentIndexName, StringComparison.Ordinal))
+    {
+        await EnsureIndexExists(currentIndexName, client, log);
+        LastIndexName = currentIndexName;
+    }
+
+    var data = new PostData<string>(myEventHubMessage);
+    var result = await client.LowLevel.IndexAsync<string>(currentIndexName, EventDocumentTypeName, data);
+
+    if (result.Success)
+    {
+        log.Info("Data successfully sent.");
+        log.Verbose(result.Body);
+    }
+    else
+    {
+        ReportError(log, $"Failed to send data.{Environment.NewLine}{result.DebugInformation}", throwException: true);
+    }
+}
+
+/// <summary>
+/// Generates an index name.
+/// </summary>
+/// <param name="indexNamePrefix">Specifies a string prefix of the index name.</param>
+/// <returns>An index name in the form of &lt;prefix&gt;-YYYY.MM.DD.</returns>
+private static string GetIndexName(string indexNamePrefix)
+{
+    DateTimeOffset now = DateTimeOffset.UtcNow;
+    string retval = $"{indexNamePrefix}-{now.Year.ToString("0000")}.{now.Month.ToString("00")}.{now.Day.ToString("00")}";
+    return retval;
+}
+
+/// <summary>
+/// Makes sure the specified index name has been created.
+/// </summary>
+/// <param name="currentIndexName">The index name to check.</param>
+/// <param name="esClient">An Elastic client instance.</param>
+/// <param name="log">A Trace writer instance.</param>
+/// <returns>A Task representing the async operation.</returns>
+private static async Task EnsureIndexExists(string currentIndexName, ElasticClient esClient, TraceWriter log)
+{
+    IExistsResponse existsResult = await esClient.IndexExistsAsync(currentIndexName);
+    if (!existsResult.IsValid)
+    {
+        ReportError(log, $"Index exists check failed.{Environment.NewLine}{existsResult.DebugInformation}", throwException: true);
+    }
+
+    if (existsResult.Exists)
+    {
+        return;
+    }
+
+    // TODO: allow the consumer to fine-tune index settings 
+    IndexState indexState = new IndexState();
+    indexState.Settings.NumberOfReplicas = 1;
+    indexState.Settings.NumberOfShards = 5;
+    indexState.Settings.Add("refresh_interval", "15s");
+
+    ICreateIndexResponse createIndexResult = await esClient.CreateIndexAsync(currentIndexName, c => c.InitializeUsing(indexState));
+
+    if (!createIndexResult.IsValid)
+    {
+        if (createIndexResult.ServerError != null &&
+            createIndexResult.ServerError.Error != null &&
+            string.Equals(createIndexResult.ServerError.Error.Type, "IndexAlreadyExistsException", StringComparison.OrdinalIgnoreCase))
```

Comment:
Minor: can use null propagation operator and avoid previous checks for null. E.g.
string.Equals(createIndexResult.ServerError?.Error?.Type, "IndexAlreadyExistsException", StringComparison.OrdinalIgnoreCase)